src/platform/Linux/CHIPLinuxStorageIni.cpp

```diff
@@ -219,97 +219,88 @@ CHIP_ERROR ChipLinuxStorageIni::GetStringValue(const char * key, char * buf, siz
 {
-    CHIP_ERROR retval            = CHIP_NO_ERROR;
     size_t encodedDataPaddingLen = 0;
     std::map<std::string, std::string> section;
-    retval = GetDefaultSection(section);
-
-    if (retval == CHIP_NO_ERROR)
+    CHIP_ERROR err = GetDefaultSection(section);
+    if (err != CHIP_NO_ERROR)
     {
-        auto it = section.find(key);
+        return err;
+    }
 
-        if (it != section.end())
-        {
-            std::string value;
+    auto it = section.find(key);
+    if (it == section.end())
+    {
+        return CHIP_ERROR_KEY_NOT_FOUND;
+    }
 
-            // Compute the expectedDecodedLen
-            if (inipp::extract(section[key], value))
-            {
-                size_t len = value.size();
+    std::string value;
 
-                encodedData                 = (char *) chip::Platform::MemoryAlloc(len + 1);
-                encodedDataLen              = value.copy(encodedData, len);
-                encodedData[encodedDataLen] = '\0';
+    // Compute the expectedDecodedLen
+    if (!inipp::extract(section[key], value))
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
 
-                // Check if encoded data was padded. Only "=" or "==" padding combinations are allowed.
-                if ((encodedDataLen > 0) && (encodedData[encodedDataLen - 1] == '='))
-                {
-                    encodedDataPaddingLen++;
-                    if ((encodedDataLen > 1) && (encodedData[encodedDataLen - 2] == '='))
-                        encodedDataPaddingLen++;
-                }
+    size_t len = value.size();
+    if (!encodedData.Alloc(len + 1))
+    {
+        return CHIP_ERROR_NO_MEMORY;
+    }
+    encodedDataLen                          = value.copy(encodedData.Ptr<char>(), len);
+    encodedData.Ptr<char>()[encodedDataLen] = '\0';
 
-                decodedDataLen = ((encodedDataLen - encodedDataPaddingLen) * 3) / 4;
-            }
-            else
-            {
-                retval = CHIP_ERROR_INVALID_ARGUMENT;
-            }
-        }
-        else
-        {
-            retval = CHIP_ERROR_KEY_NOT_FOUND;
-        }
+    // Check if encoded data was padded. Only "=" or "==" padding combinations are allowed.
+    if ((encodedDataLen > 0) && (encodedData.Ptr<char>()[encodedDataLen - 1] == '='))
+    {
+        encodedDataPaddingLen++;
+        if ((encodedDataLen > 1) && (encodedData.Ptr<char>()[encodedDataLen - 2] == '='))
+            encodedDataPaddingLen++;
     }
 
-    return retval;
+    decodedDataLen = ((encodedDataLen - encodedDataPaddingLen) * 3) / 4;
+
+    return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR ChipLinuxStorageIni::GetBinaryBlobValue(const char * key, uint8_t * decodedData, size_t bufSize, size_t & decodedDataLen)
 {
-    CHIP_ERROR retval  = CHIP_NO_ERROR;
-    char * encodedData = nullptr;
+    CHIP_ERROR retval = CHIP_NO_ERROR;
+    chip::Platform::ScopedMemoryBuffer encodedData;
```

Comment:
This is inside `namespace chip`?

Reply:
You would think so, but Inet and storage have different ones :(

```
../../src/platform/Linux/CHIPLinuxStorageIni.cpp:222:79: error: reference to ‘Platform’ is ambiguous
  222 | CHIP_ERROR ChipLinuxStorageIni::GetBinaryBlobDataAndLengths(const char * key, Platform::ScopedMemoryBuffer & encodedData,
      |                                                                               ^~~~~~~~
In file included from ../../src/lib/core/CHIPCore.h:39,
                 from ../../src/include/platform/CHIPDeviceLayer.h:25,
                 from ../../src/include/platform/internal/CHIPDeviceLayerInternal.h:23,
                 from ../../src/platform/Linux/CHIPLinuxStorageIni.cpp:31:
../../src/inet/InetLayer.h:111:11: note: candidates are: ‘namespace chip::Inet::Platform { }’
  111 | namespace Platform {
      |           ^~~~~~~~
In file included from ../../src/platform/Linux/CHIPLinuxStorageIni.h:30,
                 from ../../src/platform/Linux/CHIPLinuxStorageIni.cpp:30:
../../src/include/platform/PersistedStorage.h:34:11: note:                 ‘namespace chip::Platform { }’
   34 | namespace Platform {
```

I don't want to refactor too much, so keeping the super specific bit.

Reply:
Yeah, I didn't know we had multiple `Platform` namespaces.  Just ignore all the namespace comments.